awx/main/models/notifications.py

```diff
@@ -201,56 +247,145 @@ def get_absolute_url(self, request=None):
 
 
 class JobNotificationMixin(object):
+    STATUS_TO_TEMPLATE_TYPE = {'succeeded': 'success',
+                               'running': 'started',
+                               'failed': 'error'}
+    # Tree of fields that can be safely referenced in a notification message
+    JOB_FIELDS_WHITELIST = ['id', 'type', 'url', 'created', 'modified', 'name', 'description', 'job_type', 'inventory',
+                            'project', 'playbook', 'forks', 'limit', 'verbosity', 'job_tags', 'force_handlers', 'skip_tags',
+                            'start_at_task', 'timeout', 'use_fact_cache', 'unified_job_template', 'launch_type', 'status',
+                            'failed', 'started', 'finished', 'elapsed', 'job_explanation', 'execution_node', 'controller_node',
+                            'result_traceback', 'event_processing_finished', 'job_template', 'allow_simultaneous', 'scm_revision',
```

Comment:
@matburt @wenottingham how do you feel about omitting `result_traceback`?  It's a raw Python traceback, and probably quite lengthly and difficult to parse.

Reply:
Good catch. I'm in favor of removing for the reasons you listed.

Reply:
I'd leave it out because it may be impossible to sanitize for secrets.

Reply:
I'm fine with this.

Reply:
Removed `result_traceback`